client/src/app/shared/guards/user-is-logged-in.guard.ts

```diff
@@ -12,7 +11,7 @@ export class UserIsLoggedInGuard implements CanActivate {
   constructor(private aai: AaiService, private router: Router) {
   }
 
-  canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): Observable<boolean | UrlTree> {
-    return this.aai.isUserLoggedIn().map(loggedIn => loggedIn ? loggedIn : this.router.parseUrl(`/login?redirect=${encodeURIComponent(state.url)}`));
+  canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): boolean | UrlTree {
+    return this.aai.userLoggedIn() ? true : this.router.parseUrl(`/login?redirect=${encodeURIComponent(state.url)}`);
```

Comment:
Is there a better way to write this?
Would this work?
```
return this.aai.userLoggedIn() || this.router.parseUrl(`/login?redirect=${encodeURIComponent(state.url)}`);
```

Reply:
I think that would work:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_OR

```
o6 = false || 'Cat'      // f || t returns "Cat"
```
 And it looks like typescript correctly identifies the type of the expression to be `boolean | UrlTree`